src/backend/joanie/core/models/products.py

```diff
@@ -246,8 +246,8 @@ def generate_invoice(self):
         order = {
             "invoice_id": reference,
             "customer": {
-                "name": self.owner.get_full_name(),
-                "address": self.owner.addresses.get(main=True).get_full_address(),
+                "name": self.owner.addresses.get(is_main=True).fullname,
+                "address": self.owner.addresses.get(is_main=True).get_full_address(),
```

Comment:
1) maybe mutualize this "get" request?
2) is there not a possibility that there is no main address and it raise a DoesNotExist exception?

Reply:
Good idea to mutualize requests.
I ensure that a `ValueError` is raised if request to get the main address failed then mark order 

But I'll improve this point during the payment integration. It is currently weird to use the main address to generate the invoice. Instead we will use the billing address provided for the payment.